xmtp_mls/src/groups/message_history.rs

```diff
@@ -118,50 +117,64 @@ impl HistoryReply {
         id: &str,
         url: &str,
         bundle_hash: Vec<u8>,
-        bundle_signing_key: [u8; 32],
-        encryption_key: [u8; 32],
+        signing_key: HistoryKeyType,
+        encryption_key: HistoryKeyType,
     ) -> Self {
         Self {
             request_id: id.into(),
             url: url.into(),
             bundle_hash,
-            bundle_signing_key,
+            signing_key,
             encryption_key,
         }
     }
 }
 
 impl From<HistoryReply> for MessageHistoryReply {
     fn from(reply: HistoryReply) -> Self {
-        let bundle_signing_key = key_to_string(reply.bundle_signing_key);
-        let encryption_key = key_to_string(reply.encryption_key);
         MessageHistoryReply {
             request_id: reply.request_id,
             url: reply.url,
             bundle_hash: reply.bundle_hash,
-            bundle_signing_key,
-            encryption_key,
+            signing_key: Some(reply.signing_key.into()),
+            encryption_key: Some(reply.encryption_key.into()),
         }
     }
 }
 
-fn new_request_id() -> String {
-    Alphanumeric.sample_string(&mut rand::thread_rng(), 24)
+#[derive(Copy, Clone, Debug, PartialEq)]
+enum HistoryKeyType {
+    Chacha20Poly1305([u8; 32]),
 }
 
-#[allow(dead_code)]
-fn new_key() -> [u8; 32] {
-    let mut key = [0u8; 32];
-    let rng = rand::thread_rng();
-    rng.sample_iter(&Alphanumeric)
-        .take(32)
-        .enumerate()
-        .for_each(|(i, c)| key[i] = c);
-    key
+impl HistoryKeyType {
+    #[allow(dead_code)]
+    fn new_chacha20_poly1305_key() -> Self {
```

Comment:
In hindsight I should have suggested `chacha20_poly1305_key_material` since it's not the key itself but the inputs, but whatever. Renaming proto fields is non-breaking, and if it's really a bother we can always change it later.

Reply:
Agree. `chacha20_poly1305_key_material` is more accurate to what this field is for. 
I suspect there may be at least one additional change to the proto before those whole feature set is finished, so I'll pick it up in the next change...